### PR TITLE
perf: use `torch.split` in replace of slicing ops in repflow

### PR DESCRIPTION
**Pull request by caic99**
The benchmark result shows an overhead introduced by slicing operators in the current implementation.
This PR replaces slicing for each tensor with a unified `torch.split` op. 
It brings a speed-up of 6.7% while improves the code readability.
Tested on OMat with 9 DPA-3 layers and batch size=auto:512.

## Summary by CodeRabbit

- **Refactor**
	- Streamlined the internal logic for processing numerical components to reduce complexity.
	- Enhanced internal validation checks related to the `bias` variable to boost overall system robustness and maintainability.
	- Updated method signatures for `optim_angle_update` and `optim_edge_update` to improve clarity and usability.

### CODE REVIEW

File: deepmd/pt/model/descriptor/repflow_layer.py [hunk 1]
```diff
@@ -450,36 +439,28 @@ def optim_edge_update(
...
             matrix, bias = self.edge_self_linear.matrix, self.edge_self_linear.bias
         else:
             raise NotImplementedError
+        assert bias is not None
+
+        node_dim = node_ebd.shape[-1]
+        edge_dim = edge_ebd.shape[-1]
         assert 2 * node_dim + edge_dim == matrix.size()[0]
+        # node_dim, node_dim, edge_dim
+        node, node_ext, edge = torch.split(matrix, node_dim)
```
[the remark this answers is not in the document]
Comment (caic99) [reply deepmd/pt/model/descriptor/repflow_layer.py:455]: It's OK to do so

> If split_size_or_sections is an integer type, then [tensor](https://pytorch.org/docs/stable/generated/torch.tensor.html#torch.tensor) will be split into equally sized chunks (if possible). Last chunk will be smaller if the tensor size along the given dimension dim is not divisible by split_size.

ref: https://pytorch.org/docs/stable/generated/torch.split.html

**Review by iProzd — COMMENTED** [review]
please modify the implements in [dpmodel](deepmd/dpmodel/descriptor/repflows.py).

File: deepmd/dpmodel/descriptor/repflows.py [hunk 2]
```diff
@@ -796,42 +796,35 @@ def optim_angle_update(
...
 
-        # nf * nloc * a_sel * a_sel * angle_dim
-        sub_angle_update = xp.matmul(
-            angle_ebd, matrix[sub_angle_idx[0] : sub_angle_idx[1], :]
+        angle_dim = angle_ebd.shape[-1]
+        node_dim = node_ebd.shape[-1]
+        edge_dim = edge_ebd.shape[-1]
+        angle_dim = angle_ebd.shape[-1]
+        node_dim = node_ebd.shape[-1]
+        edge_dim = edge_ebd.shape[-1]
```
Comment (caic99) [inline deepmd/dpmodel/descriptor/repflows.py:815]:
```suggestion
```

File: deepmd/dpmodel/descriptor/repflows.py [hunk 3]
```diff
@@ -796,42 +796,36 @@ def optim_angle_update(
...
         if feat == "edge":
+            assert self.edge_angle_linear1 is not None
             matrix, bias = self.edge_angle_linear1.w, self.edge_angle_linear1.b
         elif feat == "angle":
+            assert self.angle_self_linear is not None
             matrix, bias = self.angle_self_linear.w, self.angle_self_linear.b
         else:
             raise NotImplementedError
-        assert angle_dim + node_dim + 2 * edge_dim == matrix.shape[0]
+        assert bias is not None
```
Comment (iProzd) [inline deepmd/dpmodel/descriptor/repflows.py:808]: why delete this assertion? Notice that assertions here are not consistent across backends.

**Review by iProzd — CHANGES_REQUESTED** [review]

**Review by iProzd — APPROVED** [review]

File: deepmd/pt/model/descriptor/repflow_layer.py [hunk 4]
```diff
@@ -451,42 +440,30 @@ def optim_edge_update(
...
+        assert bias is not None
 
-        # nf * nloc * node/edge_dim
-        sub_node_update = torch.matmul(
-            node_ebd, matrix[sub_node_idx[0] : sub_node_idx[1]]
-        )
+        node_dim = node_ebd.shape[-1]
+        edge_dim = edge_ebd.shape[-1]
+        # node_dim, node_dim, edge_dim
+        node, node_ext, edge = torch.split(matrix, [node_dim, node_dim, edge_dim])
```
[the remark this answers is not in the document]
Comment (caic99) [reply deepmd/pt/model/descriptor/repflow_layer.py:454]:
```suggestion
        # `torch.split` guarantees the slicing shape matches with the input shape.
        node, node_ext, edge = torch.split(matrix, [node_dim, node_dim, edge_dim])
```
